Contents/Code/__init__.py

```diff
@@ -72,51 +85,126 @@ class ShokoCommonAgent:
+                filename = urllib.unquote(media.filename)
+
+                Log('Searching movie %s - %s' % (name, filename))
+
+                # Get file data using filename
+                # http://127.0.0.1:8111/api/v3/File/PathEndsWith/%5Bjoseole99%5D%20Clannad%20-%2001%20(1280x720%20Blu-ray%20H264)%20%5B8E128DF5%5D.mkv
+                file_data = HttpReq('api/v3/File/PathEndsWith/%s' % filename)
+
+                # Take the first file. As we are searching with both parent folder and filename, there should be only one result.
+                if len(file_data) > 1:
+                    Log('File search has more than 1 result. HOW DID YOU DO IT?')
+                file_data = file_data[0]
+
+                # Get series data
+                series_id = file_data['SeriesIDs'][0]['SeriesID']['ID'] # Taking the first matching anime. Not supporting multi-anime linked files for now. eg. Those two Toradora/One Piece episodes
+                series_data = {}
+                series_data['shoko'] = HttpReq('api/v3/Series/%s' % series_id) # http://127.0.0.1:8111/api/v3/Series/24
+                series_data['anidb'] = HttpReq('api/v3/Series/%s/AniDB' % series_id) # http://127.0.0.1:8111/api/v3/Series/24/AniDB
+
+                # Get episode data
+                ep_id = file_data['SeriesIDs'][0]['EpisodeIDs'][0]['ID'] # Taking the first
+                ep_data = {}
+                ep_data['anidb'] = HttpReq('api/v3/Episode/%s/AniDB' % ep_id) # http://127.0.0.1:8111/api/v3/Episode/212/AniDB
+
+                # Make a dict of language -> title for all titles in anidb data
+                ep_titles = {}
+                for item in ep_data['anidb']['Titles']:
+                    ep_titles[item['Language']] = item['Name']
+
+                # Get episode title according to the preference
+                title = None
+                for lang in Prefs['EpisodeTitleLanguagePreference'].split(','):
+                    lang = lang.strip()
+                    title = ep_titles[lang.upper()]
+                    if title is not None: break
+                if title is None: title = ep_titles['EN'] # If not found, fallback to EN title
+                full_title = series_data['shoko']['Name'] + ' - ' + title
+
+                # Get year from air date
+                air_date = try_get(ep_data['anidb'], 'AirDate', None)
+                year = air_date.split('-')[0] if air_date is not None else None
+
+                score = 100 if series_data['shoko']['Name'] == name else 85  # TODO: Improve this to respect synonyms./
+
+                meta = MetadataSearchResult(str(ep_id), full_title, year, score, lang)
                 results.Append(meta)
 
             else: # For manual searches
-                prelimresults = HttpReq("api/serie/search?query=%s&level=%d&fuzzy=%d&ismovie=1" % (urllib.quote(name.encode('utf8')), 2, Prefs['Fuzzy']))
```

Comment:
Due to the use of Fuzzy in troubleshooting, this is going to be a blocking change,

Reply:
Added the option for non-fuzzy search